packages/plugin-npm-cli/sources/commands/npm/publish.ts

```diff
@@ -166,6 +168,7 @@ export default class NpmPublishCommand extends BaseCommand {
             ident,
             otp: this.otp,
             jsonResponse: true,
+            allowOidc: Boolean(env.CI && (env.GITHUB_ACTIONS || env.GITLAB)),
```

Comment:
Do we gain something to exposing that as an option? Shouldn't it be an implementation detail of npmHttpUtils, since only `put` needs it and it always wants to set it if possible?

Reply:
Theoretically, it could be supported anywhere related to the registry if the registry provider wanted it. It's not necessarily coupled with publish or put actions.

Since this essentially involves more than two provider details (registry and runner environment), I didn't think plugin-npm was the ideal place to do it, but I wanted to avoid too many changes here.

Another approach would be to use the `getNpmAuthenticationHeader` hook; perhaps a plugin for each CI provider would be ideal.